feat(r/burn_alert): add multi-dataset support (#644)
honeycombio/terraform-provider-honeycombio: docs/resources/burn_alert.md

Burn alerts can be imported using a combination of the dataset name and their ID, e.g.

> For multi-dataset burn alerts, replace dataset name with `__all__`.

Author: good point!

> 
> Burn alerts can be imported using a combination of the dataset name and their ID, e.g.
> 
> For multi-dataset burn alerts, replace dataset name with `__all__`.
> 
> ```
> $ terraform import honeycombio_burn_alert.my_alert my-dataset/bj9BwOb1uKz
> ```

Reviewer: ```suggestion

Burn alerts can be imported using by using their ID combined with their dataset.
If the burn alert is a multi-dataset (MD) burn alert, the dataset is not provided.

### Burn Alert

```
$ terraform import honeycombio_burn_alert.my_alert my-dataset/bj9BwOb1uKz
```

### Multi-dataset Burn Alert

```
$ terraform import honeycombio_burn_alert.my_alert bc9XwOb2yJu
```

```
